great_expectations/datasource/fluent/sources.py

```diff
@@ -519,6 +519,7 @@ def update_datasource(
             if id_:
                 updated_datasource.id = id_
 
+            updated_datasource._data_context = self._data_context
             updated_datasource.test_connection()
```

Comment:
Let's put a comment here detailing why we need to attach the context right before the test connection call? Feels like such a small nuance that can cause things to break (like it just did).

Or perhaps we encapsulate these two lines in a function to provide more context

Reply:
I plan to followup this up and move these to a `attach_context_and_test_connection()` function.